src/js/table.js

```diff
@@ -61,14 +61,12 @@ export class RegularTableViewModel {
         }
     }
 
-    async draw(container_size, view_cache, selected_id, preserve_width, viewport) {
+    async draw(container_size, view_cache, selected_id, preserve_width, viewport, num_columns) {
         const {width: container_width, height: container_height} = container_size;
-        const {view, config, column_paths, schema, table_schema} = view_cache;
-        const visible_columns = column_paths.slice(viewport.start_col);
-        const columns_data = await view.to_columns(viewport);
+        const {view, config, schema} = view_cache;
+        const {data, row_indices, column_indices, __id_column: id_column} = await view(viewport.start_col, viewport.start_row, viewport.end_col, viewport.end_row);
```

Comment:
what was the motivation behind rearranging the indices to this particular order?